File: 0101-symmetric-tree/0101-symmetric-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool ischk(TreeNode *a,TreeNode* b){
        if(a==nullptr && b ==nullptr ){return true;}
        if(a==nullptr || b==nullptr){return false;}
        if(a->val != b->val)return false;
        return ischk(a->left , b->right) && ischk(a->right,b->left);

    }
    bool isSymmetric(TreeNode* root) {
        return ischk(root->left,root->right);
    }
};

File: 0101-symmetric-tree/0101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True
        return self.isMirror(root.left, root.right)
    
    def isMirror(self, left: Optional[TreeNode], right: Optional[TreeNode]) -> bool:
        if not left or not right:
            return left is right        
        if left.val != right.val:
            return False
        return (self.isMirror(left.left, right.right) and
                self.isMirror(left.right, right.left))